pypesto/visualize/ordinal_categories.py

```diff
@@ -122,8 +139,10 @@ def plot_categories_from_inner_result(
     results: List[Dict],
     simulation: List[np.ndarray],
     timepoints: List[np.ndarray],
+    observable_ids: List[str] = None,
     condition_ids: List[str] = None,
-    petab_condition_ordering: List[str] = None,
+    condition_ids_from_petab: List[str] = None,
+    axes: Optional[plt.Axes] = None,
     **kwargs,
 ):
     """Plot the inner solutions.
```

Comment:
600 lines for a single function is definitely too much. please refactor.

Reply:
Yep... Has to be made nicer. Will do